feat(curriculum): New steps and tests for expense tracker (#52570)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/20-upcoming-python/learn-lambda-functions-by-building-a-expense-tracker/657e37c417b3b97538e30bf2.md

---
id: 657e37c417b3b97538e30bf2
title: Step 2
challengeType: 20
dashedName: step-2
---

# --description--

> Now let's talk about dictionaries in Python. A dictionary is a collection of key-value pairs. Imagine a real dictionary where you look up a word (the key) to find its definition (the value). In Python, a dictionary is made with curly braces `{}`, with key-value pairs separated by commas. Each key is followed by a colon `:`, then the value.

Reviewer: ```suggestion
Now let's talk about dictionaries in Python. A dictionary is a collection of key-value pairs. Imagine a real dictionary where you look up a word (the key) to find its definition (the value). In Python, a dictionary is defined with curly braces `{}` that contain key-value pairs separated by commas. Each key is followed by a colon `:` and the value.
```

> In our project, each expense will be stored as a dictionary with two pieces of information: `amount` and `category`. For instance, `{"amount": 50.0, "category": "Food"}` represents an expense of $50.0 in the "Food" category. Your task is to create a dictionary representing an expense.

Reviewer: ```suggestion
In your project, each expense will be stored as a dictionary with two pieces of information: `amount` and `category`. For instance, `{"amount": 50.0, "category": "Food"}` represents an expense of $50.0 in the "Food" category. Your task is to create a dictionary representing an expense.
```

> Create a dictionary named `single_expense` with two key-value pairs. The keys should be `amount` and `category`, with the values being `40.0` and `"Groceries"`, respectively.

Reviewer: ```suggestion
Define a dictionary named `single_expense` with two key-value pairs. The keys should be `amount` and `category` and the values `40.0` and `"Groceries"`, respectively.
```

# --hints--

Your dictionary should be named `single_expense`.

```js
({ test: () => assert(__userGlobals.has("single_expense")) })

```

> The single_expense dictionary should have a key named `amount` with the value `40.0`.

Reviewer: ```suggestion
The `single_expense` dictionary should have a key named `amount` with the value `40.0`.
```

```js
({ test: () => assert(__pyodide.runPython(`
    single_expense = __locals.get("single_expense")
    test_result = single_expense.get('amount') == 40.0
    test_result
`))
})

```

> The single_expense dictionary should have a key named `category` with the value `"Groceries"`.

Reviewer: ```suggestion
The `single_expense` dictionary should have a key named `category` with the value `"Groceries"`.
```

```js
({ test: () => assert(__pyodide.runPython(`
    single_expense = __locals.get("single_expense")
    test_result = single_expense.get('category') == 'Groceries'
    test_result
`))
})

```

# --seed--

## --seed-contents--

```py
--fcc-editable-region--

> # Create a dictionary named 'single_expense' below

Reviewer: ```suggestion
# Define a dictionary named 'single_expense' below
```
